internal/graph/check_fast_path.go

```diff
@@ -561,7 +561,7 @@ func (c *LocalChecker) resolveFastPath(ctx context.Context, leftChans []chan *it
 		return nil, ctx.Err()
 	case r, ok := <-rightChan:
 		if !ok {
-			return res, nil
+			return res, ctx.Err()
```

Comment:
I like this. Feels like we should be doing this change everywhere within Check API.